api/v1alpha1/openapi.yaml

```diff
@@ -2181,6 +2181,136 @@ components:
+          Systemd: '#/components/schemas/ConfigHookActionSystemdSpec'
+          Executable: '#/components/schemas/ConfigHookActionExecutableSpec'
+    ConfigHookActionSpec:
+      type: object
+      properties:
+        type:
+          type: string
+        triggerOn:
+          type: array
+          items:
+            $ref: '#/components/schemas/FileOperation'
+        timeout:
+             type: string
+             pattern: '^[1-9]\d*[smhd]$'
+             description: "Duration after which the action will be terminated. Format: number followed by 's' for seconds, 'm' for minutes, 'h' for hours, 'd' for days. Must be a positive integer."
+      required:
+        - type
+        - triggerOn
+    ConfigHookActionSystemdSpec:
+       allOf:
+        - $ref: '#/components/schemas/ConfigHookActionSpec'
+        - type: object
+          properties:
+            unit:
+              $ref: '#/components/schemas/ConfigHookActionSystemdUnit'
+          discriminator:
+            propertyName: type
+          required:
+            - unit
+    ConfigHookActionSystemdUnit:
+      type: object
+      properties:
+        name:
+          type: string
+          description: 'The name of the systemd unit on which the specified operations will be performed. This should be the exact name of the unit file, such as example.service. If the name is not populated the name will be auto discovered from the file path.'
+        operations:
+          type: array
+          items:
+            type: string
+            enum:
+              - "Enable"
+              - "Disable"
+              - "Start"
+              - "Stop"
+              - "Restart"
+              - "Reload"
+              - "DaemonReload"
+            x-enum-varnames:
+              - "SystemdEnable"
+              - "SystemdDisable"
+              - "SystemdStart"
+              - "SystemdStop"
+              - "SystemdRestart"
+              - "SystemdReload"
+              - "SystemdDaemonReload"
+          description: 'The specific systemd operations to perform on the specified unit.'
+        workDir:
+            type: string
+            description: 'The directory in which the executable will be run from if it is left empty it will run from the users home directory.'
+      required:
+        - name
+        - operations 
+    ConfigHookActionExecutableSpec:
+      allOf:
+        - $ref: '#/components/schemas/ConfigHookActionSpec'
+        - type: object
+          properties:
+            executable:
+              $ref: '#/components/schemas/ConfigHookActionExecutable'
+          discriminator:
+            propertyName: type
+          required:
+            - executable
+    ConfigHookActionExecutable:
+          type: object
+          properties:
+            command:
+              type: string
+              description: 'The path or name of the executable file to run. This can be the name of a binary located in $PATH, or a full path to the binary.'
+            args:
```

Comment:
can args be used to add an environment variable to the executable command? In order to apply a set of manifests on MicroShift using the OC client, we should be able to point at the kubeconfig file. Something like this:

`KUBECONFIG=/var/lib/microshift/resources/kubeadmin/kubeconfig oc apply -f /var/usr/acm-manifests/`

Reply:
well we can pass env directly so I will add that good point